File: android/app/src/debug/java/com/laneshadow/sandbox/LaneShadowThemeController.kt
// native-sandbox: configured
package com.laneshadow.sandbox

import com.nativesandbox.theming.ThemeController
import com.nativesandbox.theming.ThemeMode

/**
 * Theme controller for LaneShadow sandbox. Supports Auto/Light/Dark toggling
 * via the sandbox's Appearance section.
 */
object LaneShadowThemeController : ThemeController {
    private var _mode: ThemeMode = ThemeMode.Auto

    override var themeMode: ThemeMode
        get() = _mode
        set(value) { _mode = value }
}

File: ios/LaneShadow/Sandbox/Theme/LaneShadowThemeController.swift
// native-sandbox: configured
import NativeSandbox
import SwiftUI

/// Theme controller for LaneShadow sandbox. Supports Auto/Light/Dark toggling
/// via the sandbox's Appearance section.
@MainActor
public final class LaneShadowThemeController: ObservableObject, ThemeController {
    public static let shared = LaneShadowThemeController()

    @Published private var mode: NativeSandbox.ThemeMode = .auto

    public var themeMode: NativeSandbox.ThemeMode {
        get { mode }
        set { mode = newValue }
    }

    private init() {}
}
